File: app/Models/Transaction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Models\Compte;
use App\Models\User;

class Transaction extends Model
{
    use HasFactory;

     protected $fillable = [
        'compte_id',
        'montant',
        'numero_destinataire',
        'code_marchand', 
        'code_distributeur',      
        'type',
        'status',
        'date_transaction',
    ];

    protected $casts = [
        'date_transaction' => 'datetime',
    ];

    // 🔗 Relations
    public function compte()
    {
        return $this->belongsTo(Compte::class);
    }

    public function user()
    {
        return $this->hasOneThrough(User::class, Compte::class);
    }
}

File: ompay_console/lib/models/transaction.dart
class Transaction {
  final int id;
  final String type;
  final double montant;
  final String status;
  final String dateTransaction;
  final String? numeroDestinataire;

  Transaction({
    required this.id,
    required this.type,
    required this.montant,
    required this.status,
    required this.dateTransaction,
    this.numeroDestinataire,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) => Transaction(
    id: json['id'],
    type: json['type'] ?? '',
    montant: double.tryParse(json['montant'].toString()) ?? 0.0,
    status: json['status'] ?? '',
    dateTransaction: json['date_transaction'] ?? '',
    numeroDestinataire: json['numero_destinataire'],
  );
}
